src/samples/PoseWithCovariance.hpp

```diff
@@ -0,0 +1,72 @@
+#ifndef BASE_SAMPLES_POSE_WITH_COVARIANCE_HPP
+#define BASE_SAMPLES_POSE_WITH_COVARIANCE_HPP
+
+#include <string>
+#include <base/Time.hpp>
+#include <base/TransformWithCovariance.hpp>
+#include <base/samples/RigidBodyState.hpp>
+
+namespace base { namespace samples {
+
+/**
+ * Represents a 3D pose with uncertainty of the frame 'object_frame_id' in the frame 'frame_id'.
+ * A frame ID shall be globally unique and correspond to a coordinate system.
+ * Following the [Rock's conventions](http://rock.opendfki.de/wiki/WikiStart/Standards)
+ * coordinate systems are right handed with X forward, Y left and Z up.
+ *
+ * The pose \f$ ^{reference}_{object}T \f$ in this structure transforms a vector \f$ v_{object} \f$
+ * from the object frame to the reference frame:
+ * \f$ v_{reference} = ^{reference}_{object}T \cdot v_{object} \f$
+ * Example of a transformation chain:
+ * \f$ ^{world}_{sensor}T = ^{world}_{body}T \cdot ^{body}_{sensor}T \f$
```

Comment:
Thanks for including the operator *. IMO is required because rock is handling transformations (not only transporting them) and the method might be very useful.

> I don't see much benefit in forwarding all those methods.

I agree.  IMO it will be overdoing it.  

I don't have further comments. :+1: